src/chainstate/coordinator/mod.rs

```diff
@@ -556,17 +622,576 @@ impl<
+                            )
+                        }
+                        AffirmationMapEntry::PoxAnchorBlockAbsent => {
+                            // network actually affirms that this anchor block
+                            // is absent.
+                            warn!("Chose PoX anchor block for reward cycle {}, but it is affirmed absent by the network", new_reward_cycle - 1; "affirmation map" => %&canonical_affirmation_map);
+                            PoxAnchorBlockStatus::SelectedAndUnknown(block_hash.clone())
+                        }
+                        AffirmationMapEntry::Nothing => {
+                            // no anchor block selected either way
+                            PoxAnchorBlockStatus::NotSelected
+                        }
+                    }
+                }
+                PoxAnchorBlockStatus::SelectedAndUnknown(ref block_hash) => {
+                    match affirmation {
+                        AffirmationMapEntry::PoxAnchorBlockPresent => {
+                            // the network affirms that this anchor block
+                            // exists, but we don't have it locally.  Stop
+                            // processing here and wait for it to arrive, via
+                            // the downloader.
+                            info!("Anchor block {} for reward cycle {} is affirmed by the network ({}), but must be downloaded", block_hash, canonical_affirmation_map, new_reward_cycle - 1);
+                            return Ok(Some(block_hash.clone()));
+                        }
+                        AffirmationMapEntry::PoxAnchorBlockAbsent => {
+                            // matches affirmation
+                            PoxAnchorBlockStatus::SelectedAndUnknown(block_hash.clone())
+                        }
+                        AffirmationMapEntry::Nothing => {
+                            // no anchor block selected either way
+                            PoxAnchorBlockStatus::NotSelected
+                        }
+                    }
+                }
+                PoxAnchorBlockStatus::NotSelected => {
+                    // no anchor block selected either way
+                    PoxAnchorBlockStatus::NotSelected
+                }
+            }
+        } else {
+            // no-op: our view of the set of anchor blocks is consistent with
+            // the canonical affirmation map, so the status of this new anchor
+            // block is whatever it was calculated to be.
+            rc_info.anchor_status.clone()
+        };
+
+        // update new status
+        debug!(
+            "Update anchor block status for reawrd cycle {} from {:?} to {:?}",
+            new_reward_cycle, &rc_info.anchor_status, &new_status
+        );
+        rc_info.anchor_status = new_status;
+        Ok(None)
+    }
+
+    /// For unaffirmed anchor blocks, determine if they should be marked as present or absent.
+    fn has_unaffirmed_pox_anchor_block(
```

Comment:
I think this method may make more sense as `fn has_stacks_block_for(&self, block_commit: LeaderBlockCommitOp)`

Also, if the `_block_commit_metadata` parameter isn't being used, it should be removed.

Reply:
Yup, removed